lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -817,8 +817,11 @@ function( $pseudo_selector ) use ( $selector ) {
 			if ( $use_root_padding ) {
 				$block_rules .= '.wp-site-blocks { padding-top: var(--wp--style--root--padding-top); padding-bottom: var(--wp--style--root--padding-bottom); }';
 				$block_rules .= '.has-global-padding { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }';
+				$block_rules .= '.has-global-padding :where(.has-global-padding) { padding-right: 0; padding-left: 0; }';
 				$block_rules .= '.has-global-padding > .alignfull { margin-right: calc(var(--wp--style--root--padding-right) * -1); margin-left: calc(var(--wp--style--root--padding-left) * -1); }';
-				$block_rules .= '.has-global-padding > .alignfull > :where([class*="wp-block-"]:not(.alignfull):not(.alignfull):not([class*="__"]),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }';
+				$block_rules .= '.has-global-padding :where(.has-global-padding) > .alignfull { margin-right: 0; margin-left: 0; }';
+				$block_rules .= '.has-global-padding > .alignfull > :where([class*="wp-block-"]:not(.alignfull):not([class*="__"]),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }';
```

Comment:
Shouldn't this also have `:not(.alignwide)` included as well? Since the `.has-global-padding` class is already applied to the parent block? 

Currently `wide` width blocks inside a group block with global padding still get 2x the proper padding. 

![CleanShot 2022-09-05 at 20 34 47@2x](https://user-images.githubusercontent.com/1813435/188524111-a80b3e66-731c-4770-980c-58d074dd5fdf.png)

Reply:
Seems like the remaining issue occurs when a wide width block is within a fullwidth block (that's not omitted from the above code).

Reply:
It's especially noticeable when you add padding to the parent group block, regardless of constrained layout or not. 

Editor: 

<img width="1001" alt="CleanShot 2022-09-05 at 20 45 55@2x" src="https://user-images.githubusercontent.com/1813435/188524701-383c9de3-d53d-449c-be9c-c00a6b55e72d.png">

Front-end: 

<img width="1082" alt="CleanShot 2022-09-05 at 20 46 21@2x" src="https://user-images.githubusercontent.com/1813435/188524729-fb3064db-1a86-401d-850e-0823c8cb31e3.png">

You can try [this snippet](https://gist.github.com/richtabor/19ceab88f73b6ad40c58d977879da13b), with Twenty twenty Three to replicate.

Reply:
Thanks for digging in on this — I know its a tricky one.

Reply:
Yeah, that padding will be applied to any block inside the nested `alignfull` block that's not also `alignfull`; you can see it also gets applied to a regular width block:
<img width="970" alt="Screen Shot 2022-09-06 at 5 48 20 pm" src="https://user-images.githubusercontent.com/8096000/188578312-d641279d-0e7b-4675-b045-e99b512332f9.png">

(the yellow Group blocks are outside the full width Group and the Media&Text ones inside it)

This is also complicated by the `box-sizing` property on the blocks that receive padding: some of them are set to `border-box` (e.g. Media&Text) and others to `content-box` (e.g. Paragraph, Image) so results vary depending on the block types.

I'm thinking we could target this extra padding rule only to children of `alignfull` blocks that _don't_ have content width set, as they're the ones that really need that padding? Whereas in blocks with content width set it's more likely to break the alignment.

Reply:
Ok I've tried ignoring children of `alignfull` blocks with content width.